File: Apps/gm-web-app/src/app/stocks/stock-recommendation-history.ts
import { StockRecommendation } from "./stock-recommendation";

export class StockRecommendationHistory {
    constructor(public systemId: string,
        public symbol: string,
        public lastUpdated: Date,
        public lastRecommendation: StockRecommendation,
        public historicalData: Array<StockRecommendation>) {

        }
}

File: Apps/gm-flutter-app/gimmillions/lib/models/stock-recommendation-history.dart
import 'package:gimmillions/models/stock-recommendation.dart';

class StockRecommendationHistory {
  DateTime lastUpdated;
  String symbol;
  String systemId;
  StockRecommendation lastRecommendation;
  List<StockRecommendation> historicalData;

  StockRecommendationHistory(
      this.systemId, this.symbol, this.lastUpdated, this.lastRecommendation, this.historicalData) {}

  factory StockRecommendationHistory.fromJson(Map<String, dynamic> json) {
    var historicalData = json['historicalData'] as List;

    return StockRecommendationHistory(
        json['systemId'],
        json['symbol'],
        DateTime.parse(json['lastUpdated']),
        StockRecommendation.fromJson(json['lastRecommendation']),
        historicalData.map((e) => StockRecommendation.fromJson(e)).toList());
  }
}
